tooling/storybook-addon/src/components/Tool.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { useCallback } from "react"
+import { useAddonState } from "@storybook/api"
+import { Icons, IconButton } from "@storybook/components"
+import { addons } from "@storybook/addons"
+import { ADDON_ID, TOOL_ID, EVENTS } from "../constants"
+
+export const Tool = () => {
+  const persistedColorMode =
+    localStorage.getItem("chakra-ui-color-mode") === "dark"
+  const [darkMode, setDarkMode] = useAddonState(ADDON_ID, persistedColorMode)
+  const channel = addons.getChannel()
+
+  const toggleMyTool = useCallback(
+    () => {
+      channel.emit(EVENTS.TOGGLE_COLOR_MODE, !darkMode ? "dark" : "light")
+      setDarkMode(!darkMode)
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
if `channel` variable is the one causing an issue, can you initialise channel inside this useCallback itself?

Reply:
What do you mean by an issue? 
The channel is not the problem, it works. 

The part that I want to improve here is how I retrieve the color mode's persisted state.
Imagine when a dev needs to change the storageManager to cookies or in a future release, Chakra UI decides to remove the localstorage key or change it.